src/server/api/admin_api.py

```diff
@@ -87,26 +111,22 @@ def get_statistics():
 
 @admin_api.route("/api/statistics", methods=["GET"])
 def list_statistics():
-    try:
-        last_execution_file = open(LOGS_PATH + "last_execution.json", "r")
-        last_execution_details = json.loads(last_execution_file.read())
-        last_execution_file.close()
+    """ Pull Last Execution stats from DB. """
+    current_app.logger.info("list_statistics() request")
+    last_execution_details = '{}'  # Empty but valid JSON
 
-    except (FileNotFoundError):
-        current_app.logger.error("last_execution.json file was missing")
-        return abort(500)
+    try:    # See Alembic Revision ID: 05e0693f8cbb for table definition
+        with engine.connect() as connection:
+            s = text("select valcol from kv_unique where keycol = 'last_execution_time';")
+            result = connection.execute(s)
+            last_execution_details  = result.fetchone()[0]
 
-    except (json.JSONDecodeError):
-        current_app.logger.error(
-            "last_execution.json could not be decoded - possible corruption"
-        )
-        return abort(500)
 
     except Exception as e:
```

Comment:
need a finally or a return statement in the exception case, or else the UI hangs

Reply:
No, it works, because `last_execution_details` is defined above at 116.  The exception code logs the error then execution continues to the `return` at 129, which returns the 'empty' value defined at 116.